### PR TITLE
Fix Chrome 55+ mobile & new pad sensitivity option

### PR DESCRIPTION
**Pull request by inuyaksa**
- fixed Chrome 55+ mobile pich/touch flickering bug #303
- fixed Chrome passive event warning #328
- add new otion "pan sensitivity"

### CODE REVIEW

**Comment by jcnventura** [comment]
I think you should have done this in 3 different PRs... Also, looking at the PR diff, in one place you're adding 3 lines of commented code, and a few lines below you're adding some trailing space.

**Comment by inuyaksa** [comment]
Sorry, what do you mean about PRs?

Feel free to use my corrections for your project. I'm using since 2 month and it works well on all configurations desktop/mobile.

**Comment by jcnventura** [comment]
PR -> Pull requests.. There's too many changes here, and they're not related.

File: dist/jquery.panzoom.js [hunk 1]
```diff
@@ -1125,6 +1128,10 @@
 			if (this.panning) {
 				return;
 			}
+
+			var type = event.type;
+			if (window.PointerEvents&&(type === 'touchstart')) return false;  // fix double events pointer/touch on Chrome >=55 #303
```
Comment (jcnventura) [inline dist/jquery.panzoom.js]: window.PointerEvents??? Is there such a thing? I think this is a typo. And you meant window.PointerEvent. See https://developers.google.com/web/updates/2016/10/pointer-events

File: dist/jquery.panzoom.js [hunk 2]
```diff
@@ -310,6 +310,9 @@
 		// Pan only when the scale is greater than minScale
 		panOnlyWhenZoomed: false,
 
+		// Pan sensitivity, min pixels to start to panning
+		panSensitivity: 4,
+
```
Comment (jcnventura) [inline dist/jquery.panzoom.js]: New functionality move to a different PR.

File: dist/jquery.panzoom.js [hunk 3]
```diff
@@ -1136,15 +1143,17 @@
 			var origPageX = +original[4];
 			var origPageY = +original[5];
 			var panOptions = { matrix: matrix, animate: 'skip' };
-			var type = event.type;
+
+			// Indicate that we are currently panning
+			//this.panning = true;
```
Comment (jcnventura) [inline dist/jquery.panzoom.js]: Dead code.. Don't add stuff like this in a PR.

File: dist/jquery.panzoom.js [hunk 4]
```diff
@@ -1136,15 +1143,17 @@
...
 
 			// Use proper events
 			if (type === 'pointerdown') {
 				moveEvent = 'pointermove';
 				endEvent = 'pointerup';
 			} else if (type === 'touchstart') {
 				moveEvent = 'touchmove';
-				endEvent = 'touchend';
+				endEvent = 'touchend';
```
Comment (jcnventura) [inline dist/jquery.panzoom.js]: Trailing whitespace.. This change should not be here.

File: dist/jquery.panzoom.js [hunk 5]
```diff
@@ -1160,9 +1169,6 @@
 			// Remove any transitions happening
 			this.transition(true);
 
-			// Indicate that we are currently panning
-			this.panning = true;
-
```
Comment (jcnventura) [inline dist/jquery.panzoom.js]: This seems to be because of the new Pan sensitivity functionality.. Move to a different PR, please.

File: dist/jquery.panzoom.js [hunk 6]
```diff
@@ -1230,6 +1237,11 @@
 					coords = e;
 				}
 
+				if (!self.panning) {
+					if ((Math.abs(coords.pageX-startPageX)<self.options.panSensitivity)&&(Math.abs(coords.pageY-startPageY)<self.options.panSensitivity)) return true;
+				}
+
+				self.panning = true;
```
Comment (jcnventura) [inline dist/jquery.panzoom.js]: Pan sensitivity -> different PR.

File: dist/jquery.panzoom.js [hunk 7]
```diff
@@ -1196,7 +1202,8 @@
 
 			var move = function(e) {
 				var coords;
-				e.preventDefault();
+				//e.preventDefault();  // chrome passive event warning https://www.chromestatus.com/features/5093566007214080  #328
+				e.stopPropagation();
```
Comment (jcnventura) [inline dist/jquery.panzoom.js]: You can add a comment that the use of stopPropagation is because preventDefault is not allowed.. But don't comment out the old code.

**Comment by jcnventura** [comment]
That reverted all changes. The pull request is now useless, so I'll close it.

**Comment by inuyaksa** [comment]
I'm working on it, you so fast and close my PR :)
I hope soon to publish 2 distinct PRs.

**Comment by timmywil** [comment]
@inuyaksa Sounds good. Thanks for the help!
